File: gradle-plugin/src/main/java/com/github/kr328/gradle/zygote/tasks/PackagesTask.java
package com.github.kr328.gradle.zygote.tasks;

import org.gradle.api.DefaultTask;
import org.gradle.api.file.DirectoryProperty;
import org.gradle.api.provider.SetProperty;
import org.gradle.api.tasks.Input;
import org.gradle.api.tasks.OutputDirectory;
import org.gradle.api.tasks.TaskAction;

import java.io.File;
import java.nio.file.Files;

public abstract class PackagesTask extends DefaultTask {
    @Input
    public abstract SetProperty<String> getPackages();

    @OutputDirectory
    public abstract DirectoryProperty getDestinationDirectory();

    @TaskAction
    public void doAction() throws Exception {
        for (final File file : getDestinationDirectory().getAsFileTree()) {
            Files.delete(file.toPath());
        }

        for (final String pkg : getPackages().get()) {
            Files.createFile(getDestinationDirectory().file(pkg).get().getAsFile().toPath());
        }
    }
}

File: gradle-plugin/src/main/java/com/github/kr328/zloader/gradle/tasks/PackagesTask.kt
package com.github.kr328.zloader.gradle.tasks

import org.gradle.api.DefaultTask
import org.gradle.api.file.DirectoryProperty
import org.gradle.api.provider.SetProperty
import org.gradle.api.tasks.Input
import org.gradle.api.tasks.OutputDirectory
import org.gradle.api.tasks.TaskAction

abstract class PackagesTask : DefaultTask() {
    @get:Input
    abstract val packages: SetProperty<String>

    @get:OutputDirectory
    abstract val destinationDir: DirectoryProperty

    @TaskAction
    fun doAction() {
        destinationDir.asFile.get().resolve("packages").apply {
            deleteRecursively()

            mkdirs()

            packages.get().forEach {
                resolve(it).createNewFile()
            }
        }
    }
}